simulation/agents.py

```diff
@@ -0,0 +1,74 @@
+import random
+
+import pandas as pd
+
+from abm import Agent
+from simulation.constants import (
+    HEATING_SYSTEM_FUEL,
+    HEATING_SYSTEM_LIFETIME_YEARS,
+    BuiltForm,
+    Epc,
+    HeatingFuel,
+    HeatingSystem,
+    OccupantType,
+    PropertyType,
+)
+from simulation.settings import HEAT_PUMP_AWARENESS
```

Comment:
I think the `settings` file needs a little bit of thought: how would we run multiple simulations on google cloud, for example? We would want multiple `settings` files, but the current implementation where we import from `simulation.settings` makes that hard to do.

I would be open to a `settings.yaml` (or similar) file that is passed through in `parse_args()`, but not sure about the best way to import those settings into the `agents.py` here. One thought would be to have a section in the yaml/toml file called "agents", and every argument there is passed into the `Household.__init__()` (e.g `heat_pump_aware_probability`. This would allow us to also easily test for (non) heat pump aware agents, since we can set that value to 0 or 1 respectively in the tests.

Reply:
>I think the settings file needs a little bit of thought: how would we run multiple simulations on google cloud, for example? We would want multiple settings files, but the current implementation where we import from simulation.settings makes that hard to do.

This is exactly why a Python file of parameters doesn't work. 

For now I think focus on we want to do and I can make it happen. I think we want to be able to:

* run a simulation with some sensible defaults
* run a simulation with some different parameters
* run lots of simulations with different parameters (a bit like gridsearch for hyperparameter optimisation)

Reply:
Let's discuss what we think are our requirements together (maybe with @soolinglim) and come back to the points around using settings files.